r/tests/testthat/test-metadata.R

```diff
@@ -63,6 +63,13 @@ test_that("R metadata is not stored for types that map to Arrow types (factor, D
   expect_null(Table$create(example_with_times[1:3])$metadata$r)
 })
 
+test_that("R metadata is not stored for ExtensionType columns", {
+  tab <- Table$create(
+    x = vctrs::new_vctr(1:5, class = "special_integer")
+  )
```

Comment:
Was checking if this works if the extensiontype is nested too, but ran into error:

```r
tab <- Table$create(
  x = data.frame(a = vctrs::new_vctr(1:5, class = "special_integer"))
)
# Error: NotImplemented: extension
```